src/Native/Runtime/windows/CoffNativeCodeManager.cpp

```diff
@@ -0,0 +1,501 @@
+//
+typedef union _UNWIND_CODE {
+    struct {
+        uint8_t CodeOffset;
+        uint8_t UnwindOp : 4;
+        uint8_t OpInfo : 4;
+    };
+
+    uint16_t FrameOffset;
+} UNWIND_CODE, *PUNWIND_CODE;
+
+//
+// Define unwind information flags.
+//
+
+#define UNW_FLAG_NHANDLER 0x0
+#define UNW_FLAG_EHANDLER 0x1
+#define UNW_FLAG_UHANDLER 0x2
+#define UNW_FLAG_CHAININFO 0x4
+
+typedef struct _UNWIND_INFO {
+    uint8_t Version : 3;
+    uint8_t Flags : 5;
+    uint8_t SizeOfProlog;
+    uint8_t CountOfUnwindCodes;
+    uint8_t FrameRegister : 4;
+    uint8_t FrameOffset : 4;
+    UNWIND_CODE UnwindCode[1];
+} UNWIND_INFO, *PUNWIND_INFO;
+
+typedef DPTR(struct _UNWIND_INFO)      PTR_UNWIND_INFO;
+typedef DPTR(union _UNWIND_CODE)       PTR_UNWIND_CODE;
+
+#endif // _TARGET_AMD64_
+
+static PTR_VOID GetUnwindDataBlob(TADDR moduleBase, PTR_RUNTIME_FUNCTION pRuntimeFunction, /* out */ size_t * pSize)
+{
+#if defined(_TARGET_AMD64_)
+    PTR_UNWIND_INFO pUnwindInfo(dac_cast<PTR_UNWIND_INFO>(moduleBase + pRuntimeFunction->UnwindInfoAddress));
+
+    size_t size = offsetof(UNWIND_INFO, UnwindCode) + sizeof(UNWIND_CODE) * pUnwindInfo->CountOfUnwindCodes;
+
+    ASSERT((pUnwindInfo->Flags & UNW_FLAG_CHAININFO) == 0);
+
+    if (pUnwindInfo->Flags & (UNW_FLAG_EHANDLER | UNW_FLAG_UHANDLER))
+    {
+        // Personality routine
+        size = ALIGN_UP(size, sizeof(DWORD)) + sizeof(DWORD);
+    }
+
+    *pSize = size;
+
+    return pUnwindInfo;
+
+#elif defined(_TARGET_ARM_)
+
+    // if this function uses packed unwind data then at least one of the two least significant bits
+    // will be non-zero.  if this is the case then there will be no xdata record to enumerate.
+    ASSERT((pRuntimeFunction->UnwindData & 0x3) == 0);
+
+    // compute the size of the unwind info
+    PTR_TADDR xdata = dac_cast<PTR_TADDR>(pRuntimeFunction->UnwindData + moduleBase);
+
+    ULONG epilogScopes = 0;
+    ULONG unwindWords = 0;
+    ULONG size = 0;
+
+    if ((xdata[0] >> 23) != 0)
+    {
+        size = 4;
+        epilogScopes = (xdata[0] >> 23) & 0x1f;
+        unwindWords = (xdata[0] >> 28) & 0x0f;
+    }
+    else
+    {
+        size = 8;
+        epilogScopes = xdata[1] & 0xffff;
+        unwindWords = (xdata[1] >> 16) & 0xff;
+    }
+
+    if (!(xdata[0] & (1 << 21)))
+        size += 4 * epilogScopes;
+
+    size += 4 * unwindWords;
+
+    if ((xdata[0] & (1 << 20)) != 0)
+    {
+        // Personality routine
+        size += 4;
+    }
+
+    *pSize = size;
+    return xdata;
+#else
+    POTABILITY_ASSERT("GetUnwindDataBlob");
+    return NULL;
+#endif
+}
+
+
+CoffNativeCodeManager::CoffNativeCodeManager(TADDR moduleBase, PTR_RUNTIME_FUNCTION pRuntimeFunctionTable, UInt32 nRuntimeFunctionTable)
+    : m_moduleBase(moduleBase), m_pRuntimeFunctionTable(pRuntimeFunctionTable), m_nRuntimeFunctionTable(nRuntimeFunctionTable)
+{
+}
+
+CoffNativeCodeManager::~CoffNativeCodeManager()
+{
+}
+
+static int LookupUnwindInfoForMethod(UInt32 RelativePc,
```

Comment:
It is how it was in the old prototype that I have reused here - fixed it to be consistent.